lib/matplotlib/dates.py

```diff
@@ -732,20 +733,105 @@ def __call__(self, x, pos=None):
 
 
 class rrulewrapper(object):
+    def __init__(self, freq, tzinfo=None, **kwargs):
+        kwargs['freq'] = freq
```

Comment:
It would be pretty weird if it could modify a passed in dictionary, since it's an expansion of the keywords. I assume that the only way there would even be a dictionary to pass in would be if you passed it like `f(**kwargs)`. Testing it out on Python 3 seems to indicate that does *not* modify the dictionary. I can try and read up on it later, though.

```python
def test_func(*args, **kwargs):
    kwargs['demo'] = 9

if __name__ == "__main__":
    kwargs = {}
    test_func(**kwargs)
    print(kwargs)    # {}

    kwargs2 = {'demo': 6}
    test_func(**kwargs)
    print(kwargs)    # {}
```